supernode/server/preheat_bridge.go

```diff
@@ -0,0 +1,154 @@
+	return EncodeResponse(rw, http.StatusCreated, resp)
+}
+
+func (s *Server) getAllPreheatTasks(ctx context.Context, rw http.ResponseWriter, req *http.Request) error {
+	tasks, err := s.PreheatMgr.GetAll(ctx)
+	if err != nil {
+		return err
+	}
+	return EncodeResponse(rw, http.StatusOK, tasks)
+}
+
+func (s *Server) getPreheatTask(ctx context.Context, rw http.ResponseWriter, req *http.Request) error {
+	id := mux.Vars(req)["id"]
+	task, err := s.PreheatMgr.Get(ctx, id)
+	if err != nil {
+		return err
+	}
+	resp := types.PreheatInfo{
+		ID:         task.ID,
+		FinishTime: strfmt.DateTime{},
+		StartTime:  strfmt.DateTime{},
+		Status:     task.Status.String(),
+	}
+	return EncodeResponse(rw, http.StatusOK, resp)
+}
+
+func (s *Server) deletePreheatTask(ctx context.Context, rw http.ResponseWriter, req *http.Request) error {
+	id := mux.Vars(req)["id"]
+	if err := s.PreheatMgr.Delete(ctx, id); err != nil {
+		return err
+	}
+	return EncodeResponse(rw, http.StatusOK, true)
+}
+
+// ---------------------------------------------------------------------------
+// helper functions
+
+type validateFunc func(registry strfmt.Registry) error
+
+func parseRequest(body io.Reader, request interface{}, validator validateFunc) error {
+	if err := json.NewDecoder(body).Decode(request); err != nil {
+		if err == io.EOF {
+			return errortypes.New(http.StatusBadRequest, "empty body")
+		}
+		return errortypes.New(http.StatusBadRequest, err.Error())
+	}
+	if validator != nil {
+		if err := validator(strfmt.NewFormats()); err != nil {
+			return errortypes.New(http.StatusBadRequest, err.Error())
+		}
+	}
+	return nil
+}
+
+// initPreheatHandlers register preheat apis
+func initPreheatHandlers(s *Server, r *mux.Router) {
+	handlers := []*HandlerSpec{
+		{Method: http.MethodPost, Path: "/preheats", HandlerFunc: s.createPreheatTask},
+		{Method: http.MethodGet, Path: "/preheats", HandlerFunc: s.getAllPreheatTasks},
+		{Method: http.MethodGet, Path: "/preheats/{id}", HandlerFunc: s.getPreheatTask},
+		{Method: http.MethodDelete, Path: "/preheats/{id}", HandlerFunc: s.deletePreheatTask},
+	}
+	// register API
+	for _, h := range handlers {
+		if h != nil {
+			r.Path(versionMatcher + h.Path).Methods(h.Method).
+				Handler(m.instrumentHandler(h.Path, postPreheatHandler(h.HandlerFunc)))
+			r.Path("/api/v1" + h.Path).Methods(h.Method).
+				Handler(m.instrumentHandler(h.Path, postPreheatHandler(h.HandlerFunc)))
+			r.Path(h.Path).Methods(h.Method).
+				Handler(m.instrumentHandler(h.Path, postPreheatHandler(h.HandlerFunc)))
+		}
+	}
+}
+
+func postPreheatHandler(h Handler) http.HandlerFunc {
+	pctx := context.Background()
+
+	return func(w http.ResponseWriter, req *http.Request) {
+		ctx, cancel := context.WithCancel(pctx)
+		defer cancel()
+
+		// Start to handle request.
+		err := h(ctx, w, req)
+		if err != nil {
+			// Handle error if request handling fails.
+			handlePreheatErrorResponse(w, err)
+		}
+		logrus.Debugf("%v err:%v", req.URL, err)
```

Comment:
I think we should not print this line if `err` is equal to nil. But the code here would print that no matter what value the `err` is.

Reply:
It's a debug level log that indicates a request has been processed and what result is. It represents the process is success if the `err` is `nil`.